src/libraries/System.Text.Json/src/System/Text/Json/Writer/RawIndentation.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace System.Text.Json
+{
+    internal readonly struct RawIndentation : IEquatable<RawIndentation>
+    {
+        private readonly Memory<byte> _bytes;
+        private readonly int? _length;
+        private readonly byte? _byte;
+
+        private const byte Null = (byte)'\0';
+
+        public readonly byte Byte => _byte ?? JsonConstants.Space;
+        public readonly ReadOnlySpan<byte> Bytes => _bytes.Span;
+        public readonly int Length => _length ?? 2;
+
+        public static RawIndentation FromString(string value) => new(value);
+
+        private RawIndentation(string value)
+        {
+            if (value is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(nameof(value));
+            }
+
+            byte[] indentBytes = Encoding.UTF8.GetBytes(value);
+            byte? indentByte = null;
+
+            foreach (byte b in indentBytes)
+            {
+                if (JsonConstants.IndentChars.IndexOf(b) is -1)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(value)} contains an invalid character. Allowed characters are space and horizontal tab.");
+                }
+
+                if (indentByte is not Null)
```

Comment:
I'm somewhat confused by this. What does this check for the null character?

Reply:
It is an optimization to merge both indent bytes parsing, one for validation against valid indentation chars and the other to detect repetition of the same indent char.